File: Problems/No.2261_K_Divisible_Elements_Subarrays/k_divisible_elements_subarrays.py
class Solution(object):
    def countDistinct(self, nums, k, p):
        """
        :type nums: List[int]
        :type k: int
        :type p: int
        :rtype: int
        """

        # parameters
        dis_cnt = 0
        dis_subarrays = set()

        # start computing
        for i in range(len(nums)):
            divisible = 0
            subarray = ""
            for j in range(i, len(nums)):
                if (nums[j]%p == 0):
                    divisible += 1
                if (divisible > k):
                    break
                subarray += (str(nums[j]) + "-")
                if subarray not in dis_subarrays:
                    dis_cnt += 1
                    dis_subarrays.add(subarray)
        
        return dis_cnt

File: Problems/No.2261_K_Divisible_Elements_Subarrays/k_divisible_elements_subarrays.cpp
class Solution {
public:
    int countDistinct(vector<int>& nums, int k, int p) {
        
        // parameters
        int dis_cnt = 0;
        std::unordered_set<std::string> dis_subarrays;

        // start computing
        for (int i=0; i<nums.size(); i++) {

            int divisible = 0;
            std::string subarray = "";
            for (int j=i; j<nums.size(); j++) {

                if (nums[j]%p == 0) divisible++;
                if (divisible > k) break;
                subarray += (std::to_string(nums[j]) + "-");
                if (dis_subarrays.count(subarray) == 0) {

                    dis_cnt++;
                    dis_subarrays.insert(subarray);

                }

            }

        }

        return dis_cnt;

    }
};
